### ISSUE TITLE
Inconsistent Behavior with Async Requests

### ISSUE THREAD

**Issue by rschmukler** [gnarroway/hato#16]
Hey! Thanks for all your work on this library. I'm currently exploring replacing aleph with it and noticed an interesting behavior.

When making a synchronous request, exceptions are thrown, however the default function for `raise` is `identity`. This means that async exceptions are returned as values.

I would expect that by default `(http/get url)` and `@(http/get url {:async? true})` would behave identically. Furthermore this will result in unexpected behavior when using libraries like manifold:

```clojure
(d/catch (http/get "https://thisdoesntexistlol.com" {:async? true})
  (fn [err]
    (log/warn err "Error making request")))
```

Is this intentional?

Thanks again!

**Comment by gnarroway** [comment]
Hi,

Thanks for catching this. You are absolutely correct and this is an unintentional bug. I've added a fix in 0.7.0-SNAPSHOT, which will be released soonish. Let me know if it works for you.

**Comment by rschmukler** [comment]
I've confirmed that this is fixed in `0.7.0-SNAPSHOT`. Thanks for the quick response and looking into it! Cheers
